packages/react/src/components/cookieConsent/useModalPlaceHolder.ts

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+
+type RefListener = (element: HTMLElement) => React.MutableRefObject<HTMLElement | undefined>;
+
+export function useModalPlaceHolder(): RefListener {
+  const placeHolderId = 'HdsCookieConsentModalPlaceholder';
+  const observedElementRef = useRef<HTMLElement | null>(null);
+  const placeHolderRef = useRef<HTMLElement | null>(null);
+  const contentObserver = useRef<ResizeObserver | null>(null);
+
+  const createPlaceholderElement = useCallback((): HTMLElement => {
+    const placeHolderElement = document.createElement('div');
+    placeHolderElement.setAttribute('data-testid', 'html-cookie-consent-placeholder');
+    placeHolderElement.setAttribute('aria-hidden', 'true');
+    placeHolderElement.setAttribute('id', placeHolderId);
+    document.body.lastChild.after(placeHolderElement);
+    return placeHolderElement;
+  }, []);
+
+  const updatePlaceHolder = useCallback(
+    (placeHolderElement: HTMLElement | null, contentElement: HTMLElement | null) => {
+      if (!placeHolderElement) {
+        return;
+      }
+      const newHeight = contentElement ? contentElement.getBoundingClientRect().height : 0;
+      // eslint-disable-next-line no-param-reassign
+      placeHolderElement.style.height = `${newHeight}px`;
+    },
+    [],
+  );
+
+  const onObservedElementChange = useCallback(() => {
+    updatePlaceHolder(placeHolderRef.current, observedElementRef.current);
+  }, [updatePlaceHolder, placeHolderRef, observedElementRef]);
+
+  const getPlaceHolderElement = useCallback(() => {
+    return document.getElementById(placeHolderId);
+  }, [placeHolderId]);
+
+  const removeContentObserver = useCallback(() => {
+    if (contentObserver.current) {
+      contentObserver.current.disconnect();
+      contentObserver.current = null;
+    }
+  }, [contentObserver]);
+
+  const addContentObserver = useCallback(
+    (observedElement: HTMLElement) => {
+      removeContentObserver();
+      contentObserver.current = new ResizeObserver(onObservedElementChange);
+      contentObserver.current.observe(observedElement);
```

Comment:
I don't have much SSR-experience. Are React.refs / useEffects run in SSR? Side-effects after a component has been rendered should probably not happen in SSR, because there is no usage for them.

Reply:
ff

> > > In storybook docs tab, the portal version causes these errors to browser console:
> > > <img alt="image" width="588" src="https://user-images.githubusercontent.com/2777633/187179801-a77ce1c2-5f6f-4275-859c-a2ec3382e70b.png">
> > 
> > 
> > Is this because docs tab render are all modals on one page?
> 
> Probably yes 🤔 . I wonder if we could add a check for if the cookie consent modal is already shown / in dom, so that then the component would not do anything? Or then we could try to disable the docs tab for the cookie consent modal. Or try to customise it through the addon-docs storybook package

Seems that the problem is having multiple elements with same id present in the dom. 

The target element for portal can already exist, if a developer has added it manually, so detecting that to check if portal should be rendered, won't work. Maybe detecting child elements of that container might work. Unless there is a child there added by the developer for some reason. 

So, it is not very easy to check, when the modal should not be rendered. This is not really a use case for the cookie consent. No-one adds it twice.

The docs page seems quite useless with the cookie consent, so maybe we could just disable it?

Reply:
We should be able to disable the docs page for the cookie consent with adding this to stories parameters:

`  docs: {
    disable: true,
  },`

Reply:
Worked!